ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthEstimateGas.java

```diff
@@ -151,6 +151,7 @@ private JsonRpcErrorResponse errorResponse(
       final TransactionProcessingResult resultTrx = result.getResult();
       if (resultTrx != null && resultTrx.getRevertReason().isPresent()) {
         jsonRpcError = JsonRpcError.REVERT_ERROR;
+        jsonRpcError.setData(resultTrx.getRevertReason().get().toHexString());
```

Comment:
Revert reason could not be present.  Try something like

`jsonRpcError.setData(resultTrx.getRevertReason().map(Bytes::toHexString()).orElse(null));`

Reply:
It's checked up there on the line 152 (`resultTrx.getRevertReason().isPresent()`)